apps/test/unit/gamelab/GameLabSpriteTest.js

```diff
@@ -700,6 +700,174 @@ describe('GameLabSprite', function () {
+      animation.update();
+      expect(animation.getFrame()).to.equal(4);
+    });
+
+    it('plays the animation backward to the target frame when target < current', function () {
+      animation.changeFrame(5);
+      animation.goToFrame(2);
+
+      // Verify state on each frame
+      expect(animation.getFrame()).to.equal(5);
+
+      animation.update();
+      expect(animation.getFrame()).to.equal(4);
+
+      animation.update();
+      expect(animation.getFrame()).to.equal(3);
+
+      animation.update();
+      expect(animation.getFrame()).to.equal(2);
+
+      // Note the animation stops at the target frame.
+      animation.update();
+      expect(animation.getFrame()).to.equal(2);
+    });
+
+    it('pauses the frame after it reaches the target frame', function () {
+      // When going forward
+      animation.changeFrame(5);
+      animation.goToFrame(7);
+      expect(animation.playing).to.be.true;
+
+      animation.update();
+      animation.update();
+      expect(animation.getFrame()).to.equal(7);
+      expect(animation.playing).to.be.true;
+
+      animation.update();
+      expect(animation.getFrame()).to.equal(7);
+      expect(animation.playing).to.be.false;
+
+      // When going backward
+      animation.changeFrame(4);
+      animation.goToFrame(2);
+      expect(animation.playing).to.be.true;
+
+      animation.update();
+      animation.update();
+      expect(animation.getFrame()).to.equal(2);
+      expect(animation.playing).to.be.true;
+
+      animation.update();
+      expect(animation.getFrame()).to.equal(2);
+      expect(animation.playing).to.be.false;
+    });
+
+    it('pauses on the next frame when target == current', function () {
+      animation.changeFrame(5);
+      animation.goToFrame(5);
+      expect(animation.playing).to.be.true;
+
+      animation.update();
+      expect(animation.playing).to.be.false;
+    });
+
+    describe('when target frame is out of bounds', function () {
+      it('does not affect play behavior', function () {
+        // Play forwards
+        animation.changeFrame(5);
+        animation.play();
+        expect(animation.getFrame()).to.equal(5);
+
+        // Verify playing forwards
+        animation.update();
+        expect(animation.getFrame()).to.equal(6);
+
+        // Try to go to a negative frame.
+        // Unless ignored, we'd expect the animation to start going backwards
+        animation.goToFrame(-1);
+        animation.update();
+        expect(animation.getFrame()).to.equal(7);
+
+        // Play backwards (correctly this time)
+        animation.goToFrame(0);
+        animation.update();
+        expect(animation.getFrame()).to.equal(6);
+
+        // Try going to a positive frame out of bounds
+        // Unless ignored, we'd expect the animation to run forward again
+        animation.goToFrame(animation.images.length);
+        animation.update();
+        expect(animation.getFrame()).to.equal(5);
```

Comment:
Just above here we call `animation.goToFrame(0)` when our current frame is 7 - this causes our animation to start playing backwards.  We do one update to verify that we're playing backwards, which puts the current frame at 6.

Then, we call `animation.goToFrame(animation.images.length)` where `length` here is 10, and valid frame indices are in the range 0-9, so it's an invalid parameter.  The expected behavior in this case is to do nothing, so we continue playing backwards - which is why we end up at frame 5.